src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -475,6 +480,12 @@ export class ObjectExplorerService implements IObjectExplorerService {
 		refresh: boolean = false): Promise<azdata.ObjectExplorerExpandInfo> {
 		let self = this;
 		return new Promise<azdata.ObjectExplorerExpandInfo>((resolve, reject) => {
+			const sessionFilterCache = this._nodeFilterCache.get(session.sessionId!);
+			// If the node has filters we need to cache them so that we can reapply them when the node is refreshed.
+			if (node.filters) {
```

Comment:
if `sessionId` is not received, `sessionFilterCache` could be left `undefined`.

```suggestion
			if (node.filters && sessionFilterCache) {
```